libpdf/textbox.py

```diff
@@ -387,7 +387,14 @@ def similarity_referee(  # pylint: disable=too-many-branches  # for readability
                 and abs(x.y1 - lt_textboxes_in_rect[title_winner_idx].y1) < CHAPTER_RECTANGLE_EXTEND
             ]
             if len(potential_chapter_number) == 1:
-                winners.append(potential_chapter_number[0])
+                # In case 5, the unexpected chapter number may be extracted. To avoid it, the potential chapter number
+                # extracted around the chapter title shall be checked if it matches the patterns of comman chapter
+                # nummber e.g. 3.9.3, XII.I.V, or A.B.D.
+                pattern = re.compile(r'^(?=\w)((^|\.)(([iIvVxX]{1,8})|[a-zA-Z]|[0-9]+))+\.?(?!.)')
```

Comment:
So the string starts with a ^ but, then there is a positive lookahead for \w. Then there is an OR of ^, \.
Does this make sense? How can there be a ^ if the string already starts with ^? Please explain the regex in the comment.

Reply:
I'd go with this `^(([a-zA-Z0-9]+|[IiVvXx]+)\.?)+ `
See here https://regex101.com/r/HkJKH5/1